.gitignore

```diff
@@ -7,9 +7,10 @@ yarn-error.log
 config/config.dev.json
 .yalc/
 yalc.lock
+package-lock.json
```

Comment:
usually, its bad idea to ignore this file, right? 

would add comment about the error that having this file causes

Reply:
@mateusoliveira43 , 

downstream MTC UI is failing scratch builds
```
Error in plugin resolve_remote_source: Cachito request is in "failed" state, reason: File check failed for yarn: the package-lock.json file must not be present
```

```
Fetching the source from "https://github.com/migtools/mig-ui" at reference "51e0ff92cb62209fdf8e7558f8334f9234231a37"
...
cachito.errors.InvalidRepoStructure: File check failed for yarn: the package-lock.json file must not be present
```

[PR#1521](https://github.com/migtools/mig-ui/pull/1521) appears to introduce a net new package-lock.json file.  Looking at its history, the last time it was present was 2019.

Reply:
So we want to gitignore this file so it doesn't get checked into the repository. It will get generated when building.